Source/Core/DolphinQt2/HotkeyScheduler.cpp

```diff
@@ -0,0 +1,354 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/HotkeyScheduler.h"
+
+#include <algorithm>
+#include <thread>
+
+#include <QCoreApplication>
+
+#include "AudioCommon/AudioCommon.h"
+#include "Common/Thread.h"
+#include "Core/Core.h"
+#include "Core/HotkeyManager.h"
+#include "Core/IOS/IOS.h"
+#include "Core/IOS/USB/Bluetooth/BTBase.h"
+#include "Core/State.h"
+#include "DolphinQt2/MainWindow.h"
+#include "DolphinQt2/Settings.h"
+#include "InputCommon/ControllerInterface/ControllerInterface.h"
+#include "VideoCommon/VertexShaderManager.h"
+#include "VideoCommon/VideoConfig.h"
+
+HotkeyScheduler::HotkeyScheduler(MainWindow* parent) : m_parent(parent)
+{
+  HotkeyManagerEmu::Initialize();
+  HotkeyManagerEmu::LoadConfig();
+  HotkeyManagerEmu::Enable(true);
+}
+
+void HotkeyScheduler::start()
+{
+  m_stop_requested = false;
+  m_thread = std::thread([this] { run(); });
+  m_thread.detach();
+}
+
+void HotkeyScheduler::stop()
+{
+  m_stop_requested = true;
+  m_thread.join();
```

Comment:
`join()` cannot be called on detached threads. [detach()](http://en.cppreference.com/w/cpp/thread/thread/detach) sets the internal joinable thread variable to false, and [join()](http://en.cppreference.com/w/cpp/thread/thread/join) throws a `std::system_error` if called when joinable is false.